allowed-breaking-changes.txt

```diff
@@ -74,3 +74,5 @@ removed:@aws-cdk/aws-stepfunctions-tasks.BatchSubmitJobProps.jobDefinition
 strengthened:@aws-cdk/aws-stepfunctions-tasks.BatchSubmitJobProps
 removed:@aws-cdk/aws-lambda-event-sources.ManagedKafkaEventSourceProps.cluster
 strengthened:@aws-cdk/aws-lambda-event-sources.ManagedKafkaEventSourceProps
+
+changed-type:@aws-cdk/aws-lambda-event-sources.S3EventSource.bucket
```

Comment:
@RomainMuller Can you confirm this is indeed needed? We changed the method to accept `IBucket` instead of `Bucket` and for some reason `jsii` isn't happy about it.

Reply:
You are narrowing the type of a property of a class. This means existing code that depends on features exclusively available on the `Bucket` class will start to fail here... And Java/C#/Go code that assigns the value of the property to a `Bucket` variable will also start failing...

So yes, this is a breaking change, and it is expected that `jsii-diff` will flag this as such.

Reply:
Ahh, makes sense.

Reply:
@michaelbrewer Damn I forgot to get rid of this change - can you remove it?

Reply:
Done